### PR TITLE
Updated old dependencies

### LINKED ISSUE #508: Error updating dependency versions
### Description
I'm trying to update dependencies of the project. My forked changes: 
[https://github.com/OmkarPh/scancode-workbench/tree/fix/oldDependencies](https://github.com/OmkarPh/scancode-workbench/tree/fix/oldDependencies)

I did npm update & npm install. Also did the electron-rebuild to ensure latest native builds for sqlite3
But am facing the following err:
image

Err stack trace:
Uncaught Error: Loading non-context-aware native module in renderer: '/Users/omkar/projects/scancode-workbench/node_modules/sqlite3/lib/binding/electron-v16.0-darwin-arm64/node_sqlite3.node'. See https://github.com/electron/electron/issues/18397.
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:5)
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1199)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:5)
    at Module.load (node:internal/modules/cjs/loader:988)
    at Module._load (node:internal/modules/cjs/loader:829)
    at Function.c._load (node:electron/js2c/asar_bundle:5)
    at Function.o._load (node:electron/js2c/renderer_init:33)
    at Module.require (node:internal/modules/cjs/loader:1012)
    at require (node:internal/modules/cjs/helpers:94)
    at Object.<anonymous> (/Users/omkar/projects/scancode-workbench/node_modules/sqlite3/lib/sqlite3-binding.js:4)

### System configuration
* Macbook air M1, MacOS Monterey 12.1
* Forked from latest develop branch

### PR DESCRIPTION
**Pull request by OmkarPh**
Fixes #508

### CODE REVIEW

**Comment by OmkarPh** [comment]
Hi @pombredanne 
How can i fix this continuous-integration/appveyor/pr check

File: assets/app/js/workbenchDB.js [hunk 1]
```diff
@@ -46,12 +47,14 @@ class WorkbenchDB {
...
-      storage: storage
+      dialectModule: sqlite3,
+      storage: storage,
+      logging: false
     });
 
     this.db = Database(this.sequelize, Sequelize);
 
-    // A promise that will return when the db and tables have been created
+    // A promise that will return when the db and tables have been created\
```
Comment (pombredanne) [inline assets/app/js/workbenchDB.js]: Note sure the trailing \ is needed?

File: assets/app/js/workbenchDB.js [hunk 2]
```diff
@@ -377,7 +380,7 @@ class WorkbenchDB {
   _addFlattenedFiles(files) {
     // Fix for issue #232
     $.each(files, (i, file) => {
-      if (file.type === 'directory' && file.hasOwnProperty('size_count')) {
+      if (file.type === 'directory' && Object.prototype.hasOwnProperty.call(file, 'size_count')) {
```
Comment (pombredanne) [inline assets/app/js/workbenchDB.js:383]: What's the rationale for the change in calling syntax?

File: assets/app/js/workbenchDB.js [hunk 3]
```diff
@@ -413,7 +416,7 @@ class WorkbenchDB {
         .then(() => this.db.Email.bulkCreate(this._addExtraFields(files, 'emails'), options))
         .then(() => this.db.Url.bulkCreate(this._addExtraFields(files, 'urls'), options))
         .then(() => this.db.ScanError.bulkCreate(this._addExtraFields(files, 'scan_errors'), options))
-        .then(() => this.sequelize.Promise.each(files, (file) => {
+        .then(() =>  this.sequelize.Promise.each(files, (file) => {
```
Comment (pombredanne) [inline assets/app/js/workbenchDB.js]: This extra space may not be needed?

File: assets/app/js/workbenchDB.js [hunk 4]
```diff
@@ -515,4 +518,4 @@ class WorkbenchDB {
   }
 }
 
-module.exports = WorkbenchDB;
+module.exports = WorkbenchDB;
```
Comment (pombredanne) [inline assets/app/js/workbenchDB.js]: You should not remove the trailing LF.

File: index.html [hunk 5]
```diff
@@ -551,6 +551,7 @@ <h4 class="modal-title">ScanCode Header Information</h4>
     </div><!-- /.modal -->
 
     <script>require('./assets/app/js/renderer.js');</script>
+    <!-- <script src='./assets/app/js/renderer.js'></script> -->
```
Comment (pombredanne) [inline index.html]: Why add this as a comment?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks!
This is looking great. I just have a few minor nits for your review.

File: assets/app/js/workbenchDB.js (hunk 2 above)
Comment (OmkarPh) [reply assets/app/js/workbenchDB.js:383]: I got the warning that calling hasOwnProperty directly on object is deprecated. Hence, changed it to the recommended syntax
Reference: 
https://stackoverflow.com/a/39283005/11768101
https://dev.to/aman_singh/what-s-the-deal-with-object-prototype-hasownproperty-call-4mbj

File: index.html (hunk 5 above)
Comment (OmkarPh) [reply index.html]: Actually, I had to experiment, add logs here and there, I removed most of the residue, but some code was left :/

File: assets/app/js/workbenchDB.js (hunk 4 above)
Comment (OmkarPh) [reply assets/app/js/workbenchDB.js]: Ok

File: assets/app/js/workbenchDB.js (hunk 3 above)
Comment (OmkarPh) [reply assets/app/js/workbenchDB.js]: My bad :)

File: assets/app/js/workbenchDB.js (hunk 1 above)
Comment (OmkarPh) [reply assets/app/js/workbenchDB.js]: another residue :)

**Comment by pombredanne** [comment]
@OmkarPh the CI needs to be entirely reworked. Possibly with GH or Azure pipelines
For the Appveyor part, there is a need to at least update the image https://github.com/nexB/scancode-workbench/blob/8b4ba385b22965d5e047ee89f62608117926ba50/appveyor.yml#L3

File: assets/app/js/workbenchDB.js (hunk 2 above)
Comment (pombredanne) [reply assets/app/js/workbenchDB.js:383]: thanks

File: assets/app/js/renderer.js [hunk 6]
```diff
@@ -442,7 +441,7 @@ $(document).ready(() => {
     });
   }
 
-  /** Export JSON file with original ScanCode data and conclusions that have been created */
+  /** Export JSON file with original ScanCode data and conclusions that have been created *
```
Comment (pombredanne) [inline assets/app/js/renderer.js]: The unclosed comment may be a problem?

**Comment by OmkarPh** [comment]
> @OmkarPh the CI needs to be entirely reworked. Possibly with GH or Azure pipelines For the Appveyor part, there is a need to at least update the image
> 
> https://github.com/nexB/scancode-workbench/blob/8b4ba385b22965d5e047ee89f62608117926ba50/appveyor.yml#L3



I've updated image & node version, but there's still an err in build.py
image

**Comment by pombredanne** [comment]
> > @OmkarPh the CI needs to be entirely reworked. Possibly with GH or Azure pipelines For the Appveyor part, there is a need to at least update the image
> > https://github.com/nexB/scancode-workbench/blob/8b4ba385b22965d5e047ee89f62608117926ba50/appveyor.yml#L3
> 
> I've updated image & node version, but there's still an err in build.py image

The python script used for packaging likely needs to be updated.... You use a very recent version of Python!
I reckon you run Windows, correct?

**Review by pombredanne — APPROVED** [review]
This is looking to me in any case. And is merge ready. The CI can be fixed separately.

**Comment by pombredanne** [comment]
The CI Python script is tracked in https://github.com/nexB/scancode-workbench/issues/502

**Comment by OmkarPh** [comment]
> > I've updated image & node version, but there's still an err in build.py image
> 
> The python script used for packaging likely needs to be updated.... You use a very recent version of Python! I reckon you run Windows, correct?

I didn't run that, they were the logs from appveylor check on this PR
I ran that build on my MacBook m1 where it works fine, Maybe something's wrong in appveylor config only
image
